### PR TITLE
More code owners

### PR DESCRIPTION
**Pull request by GiudGiud**
See https://github.com/idaholab/moose/pull/18059

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -29,12 +29,13 @@
 /framework/*/utils/PetscSupport* @fdkong
 /framework/*/utils/SlepcSupport* @fdkong
 /framework/*/variables @rwcarlsen
+/framework/*/executioners/*Solve* @GiudGiud
```
Comment (loganharbour) [inline CODEOWNERS:32]: Come on man... it's in alphabetical order

**Review by loganharbour — CHANGES_REQUESTED** [review]

**Review by loganharbour — DISMISSED** [review]

**Comment by GiudGiud** [comment]
let s wait a bit to see if @bwspenc @roystgnr @jiangwen84 @joshuahansel or @dschwen want to join in on this

**Comment by roystgnr** [comment]
The pattern globbing here is limited to filenames, yes?  I'd love something that screams at me any time someone modifies a line matching "DistributedMesh" or possibly even just "distributed", but I don't think there's particular whole-file-matches I'd want to tag myself on.

File: CODEOWNERS [hunk 2]
```diff
@@ -34,8 +35,9 @@
 /modules/external_petsc_solver @fdkong
 /modules/fluid_properties @rwcarlsen @andrsd
 /modules/fsi @fdkong
-/modules/navier_stokes @lindsayad
+/modules/navier_stokes @lindsayad @GiudGiud
```
Comment (dschwen) [inline CODEOWNERS:38]:
```suggestion
/modules/navier_stokes @lindsayad @GiudGiud
/modules/phase_field @dschwen
```

Comment (dschwen) [reply CODEOWNERS:38]: Also put me in for 

```
/framework/*/utils/Units* @dschwen
/framework/*/materials/Derivative* @dschwen
/framework/*/materials/Parsed* @dschwen
```

Thanks!`

**Comment by loganharbour** [comment]
@GiudGiud get rid of that merge commit... come on

**Comment by loganharbour** [comment]
> The pattern globbing here is limited to filenames, yes?

Correct, unfortunately.

**Review by loganharbour — DISMISSED** [review]

**Comment by GiudGiud** [comment]
I think we re good to merge now?

**Review by loganharbour — APPROVED** [review]

**Pull request merged**